components/http/okHttp/src/main/java/com/microsoft/kiota/http/OkHttpRequestAdapter.java

```diff
@@ -611,24 +614,29 @@ private Response throwIfFailedResponse(@Nonnull final Response response, @Nonnul
                 if(rootNode == null) {
 		            spanForAttributes.setAttribute(errorBodyFoundAttributeName, false);
                     closeResponse = false;
-                    final ApiException result = new ApiException("service returned status code" + statusCode + " but no response body was found");
-                    result.responseStatusCode = statusCode;
-                    result.setResponseHeaders(responseHeaders);
+                    final ApiException result = new ApiExceptionBuilder()
+                            .withMessage("service returned status code" + statusCode + " but no response body was found")
+                            .withResponseStatusCode(statusCode)
+                            .withResponseHeaders(responseHeaders)
+                            .build();
                     spanForAttributes.recordException(result);
                     throw result;
                 }
                 spanForAttributes.setAttribute(errorBodyFoundAttributeName, true);
                 final Span deserializationSpan = GlobalOpenTelemetry.getTracer(obsOptions.getTracerInstrumentationName()).spanBuilder("getObjectValue").setParent(Context.current().with(span)).startSpan();
                 try(final Scope deserializationScope = deserializationSpan.makeCurrent()) {
                     final Parsable error = rootNode.getObjectValue(errorClass);
-                    ApiException result;
+                    ApiExceptionBuilder resultBuilder;
                     if (error instanceof ApiException) {
-                        result = (ApiException)error;
+                        resultBuilder = new ApiExceptionBuilder((ApiException)error);
```

Comment:
I'm not sure about that last change. We'd effectively be wrapping all exceptions in a parent Api exception. Since all exceptions inherit from API exception, this seems superfluous, and will create confusion: which object contains the headers and status code from a consumer perspective?

Reply:
> We'd effectively be wrapping all exceptions in a parent Api exception.

why? here we only pass through a Builder pattern to keep the final object immutable for the user, that's it.
This change is mostly cosmetics and doesn't alter the original behavior (AFAIU).

> which object contains the headers and status code from a consumer perspective?

The final `ApiException` returned to the user.

Have I missed any invariant here?

Reply:
right, I missed a subtility in the builder implementation. Disregard.